index.js

```diff
@@ -103,13 +103,19 @@ export const getAvailablePurchases = () => Platform.select({
  * @returns {Promise<SubscriptionPurchase>}
  */
 export const buySubscription = (sku, oldSku, prorationMode) => {
-  return Platform.select({
-    ios: async() => RNIapIos.buyProduct(sku),
-    android: async() => {
-      if (!prorationMode) prorationMode = -1;
-      return RNIapModule.buyItemByType(ANDROID_ITEM_TYPE_SUBSCRIPTION, sku, oldSku, prorationMode);
-    },
-  })();
+    return new Promise((resolve, reject) => {
```

Comment:
Looks redundant. Could you just change it this way?
```
buySubscription = async (sku, oldSku, prorationMode) => {
```

wrapping with async function will just resolve the promises.
Could you first check this?